### PR TITLE
feat: return value from UseRequestContext callback

### PR DESCRIPTION
**Pull request by TLmaK0**
Fixes: #21

### CODE REVIEW

**Comment by B4nan** [comment]
thanks, can you please add some basic tests to verify it works and ensure we dont break it?

**Comment by TLmaK0** [comment]
Yes, of course, I will try next week

File: tests/mikro-orm.common.spec.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import {UseRequestContext, MikroOrmModule} from "../src";
```
Comment (B4nan) [inline tests/mikro-orm.common.spec.ts:1]: please follow the code style that this project is using. (single quotes, spaces in import statements)

File: src/mikro-orm.common.ts [hunk 2]
```diff
@@ -18,9 +18,13 @@ export function UseRequestContext() {
...
 
+      let result;
+
       await RequestContext.createAsync(context.orm.em, async () => {
-        await originalMethod.apply(context, args);
+        result = await originalMethod.apply(context, args);
       });
+
+      return result;
     };
```
Comment (Langstra) [inline src/mikro-orm.common.ts:28]: Can't you rewrite this to something like:
```
return RequestContext.createAsync(context.orm.em, () => {
        return originalMethod.apply(context, args);
});
```

File: tests/mikro-orm.common.spec.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  baseDir: __dirname,
+  entities: ['entities'],
+};
+
+const TEST_VALUE='expected value';
+
+@Injectable()
+class TestClass {
+  constructor(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars-experimental
```
Comment (Langstra) [inline tests/mikro-orm.common.spec.ts:18]: Suppose that if this line it unused that it can be removed?

File: src/mikro-orm.common.ts (hunk 2 above)
Comment (TLmaK0) [reply src/mikro-orm.common.ts:28]: I didn't find a way to do this, as SimoneDagma points here https://github.com/mikro-orm/nestjs/issues/21#issuecomment-814779527 the `RequestContext.createAsync` returns `Promise<void>`

File: tests/mikro-orm.module.spec.ts [hunk 4]
```diff
@@ -29,12 +29,12 @@ class ConfigService implements MikroOrmOptionsFactory {
 @Module({ providers: [ConfigService, myLoggerProvider], exports: [ConfigService] })
 class ConfigModule { }
 
-const getEntityManagerLoop = async (module: TestingModule): Promise<Set<string>> => {
+const getEntityManagerLoop = async (module: TestingModule): Promise<Set<number | string>> => {
```
Comment (B4nan) [inline tests/mikro-orm.module.spec.ts:32]:
```suggestion
const getEntityManagerLoop = async (module: TestingModule): Promise<Set<string>> => {
```

File: tests/mikro-orm.module.spec.ts [hunk 5]
```diff
@@ -29,12 +29,12 @@ class ConfigService implements MikroOrmOptionsFactory {
...
   // this function mocks the contextId factory which is called on each request
   // it's looped 5 times and resolves the EntityManager provider 10 times
   // set only allows unique values, it should only return 5 items as it should resolve the same em with the same contextId
 
-  const generatedIds = new Set<string>();
+  const generatedIds = new Set<number | string>();
```
Comment (B4nan) [inline tests/mikro-orm.module.spec.ts:37]:
```suggestion
  const generatedIds = new Set<string>();
```

**Review by B4nan — APPROVED** [review]

**Comment by B4nan** [comment]
looks like i dont have write access to your fork, so please either rebase this and remove the conflicting changes in `tests/mikro-orm.module.spec.ts`

**Comment by B4nan** [comment]
thanks!

**Comment by TLmaK0** [comment]
My pleasure

**Pull request merged**